sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -553,6 +553,59 @@ def test_ordinal_encoder_raise_missing(X):
         ohe.transform(X)
 
 
+def test_ordinal_encoder_handle_unknowns_string():
+    enc = OrdinalEncoder(handle_unknown='use_encoded_value', unknown_value=-2)
```

Comment:
we should test that `unknown_value=0` or `1` is also valid (but not for `inverse_transform`).

Reply:
Not sure what you mean. Using `unknown_value=0` or `1` would usually lead to an error during fit, because these are values already used for encoding the seen categories.

Reply:
Why would that be an error? What's wrong with the user wanting to conflate unknowns with a specific known category? I had thought we would want to handle that case. If we won't allow that case then we should certainly test that an error is raised if it is attempted.

Reply:
There is a test already for checking correct raise in case of using an `unknown_value` that is already used for encoding one of the categories in the fit.

Reply:
And nothing wrong with conflating a specific known category with unknowns. I think it is just a matter of taste whether we want to allow conflating or not. However, I would suggest to allow it only in mode `use_category` (Which we said we would include in a later version to keep it small for now, see [#16959](https://github.com/scikit-learn/scikit-learn/pull/16959#issuecomment-657311133).), not in `use_encoded_value`, because it would be clearer to explicitly state the category to be conflated. But I am fine to allow it here for `use_encoded_value` as well, if you prefer that (It just means deleting a few lines of code ;).).

Reply:
TBH, I don't mind whether we allow or disallow it at this point, but we need to explicitly do one or the other, and test it.

Reply:
Agreed. Please let me know if the test I added for this (last check in `test_ordinal_encoder_handle_unknowns_raise`) is not sufficient.

Reply:
@thomasjpfan can you confirm whether it was your expectation to allow/disallow `unknown_value=0`?